src/variety/pencils.js

```diff
@@ -0,0 +1,804 @@
+//
+// pencils.js: Implementation of Pencils puzzle type.
+//
+(function(pidlist, classbase){
+	if(typeof module==='object' && module.exports){module.exports = [pidlist, classbase];}
+	else{ pzpr.classmgr.makeCustom(pidlist, classbase);}
+}(
+['pencils'], {
+//---------------------------------------------------------
+// マウス入力系
+MouseEvent:{
+	inputModes:{edit:['arrow','number','undef','clear'],play:['border','line','arrow','peke','bgcolor','bgcolor1','bgcolor2']},
+	mouseinput_number : function(){
+		if(this.mousestart){ this.inputqnum_pencils();}
+	},
+	mouseinput : function(){ // オーバーライド
+		if(this.inputMode==='undef'){
+			if(this.mousestart){ this.inputqnum_pencils();}
+		}
+		else{ this.common.mouseinput.call(this);}
+	},
+	mouseinput_auto : function(){
+		if(this.puzzle.playmode){
+			if(this.mousestart || this.mousemove){
+				if(this.btn==='left'){
+					if(this.isBorderMode()) {
+						this.inputborder();
+					} else{
+						this.inputLineOrArrow();
+					}
+				} else {
+					this.inputpeke();
+				}
+			}
+			else if(this.mouseend && this.notInputted() && !this.isBorderMode()) {
+				this.inputBGcolor();
+			}
+		}
+		else if(this.puzzle.editmode){
+			if(this.mousestart || this.mousemove){ this.inputarrow_cell();}
+			else if(this.mouseend && this.notInputted()){ this.inputqnum_pencils();}
+		}
+	},
+
+	inputLineOrArrow : function(){
```

Comment:
Just from playing with the solving interface a bit, I feel it might be more expected if dragging from a number placed an arrow even if there's no border yet (and started drawing a line after that). Would that make sense?

Reply:
This situation feels too specific. From a discoverability standpoint, it doesn't feel right to add another way to input arrows which is 1) easier to find than the current method, and 2) insufficient in most situations. If you already know the length and direction of a pencil, you can just draw the outline first, then draw the arrow and line afterwards.

It makes more sense to me to prevent drawing lines through numbers, either in Auto mode or globally using a prehook.

Reply:
That's fair, I think I agree it would be more confusing than anything. I think this came when I was solving a puzzle with clues at the end of the pencil, but of course that's not necessarily the case.